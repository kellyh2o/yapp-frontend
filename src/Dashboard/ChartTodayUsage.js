import React from 'react';
import { connect } from "react-redux";
import { useTheme } from '@material-ui/core/styles';
import { LineChart, Line, XAxis, YAxis, Label, ResponsiveContainer } from 'recharts';
import Title from './Title';

// Generate Sales Data
function createData(time, amount) {
  return { time, amount };
}

const ChartTodayUsage = (props) => {
    const theme = useTheme();

    let jumpData = [];

    if (props.demoLocation && props.demoLocation.jumps) {
        const jumps = props.demoLocation.jumps;
        const events = [];
        
        jumps.forEach(jump => {
            jump.events.forEach(event => {
                events.push(event);
            });
        });

        const now = new Date();
        const startOfDay = new Date(now.getFullYear(), now.getMonth(), now.getDate());
        const hour3 = new Date(now.getFullYear(), now.getMonth(), now.getDate(), 3);
        const hour6 = new Date(now.getFullYear(), now.getMonth(), now.getDate(), 6);
        const hour9 = new Date(now.getFullYear(), now.getMonth(), now.getDate(), 9);
        const hour12 = new Date(now.getFullYear(), now.getMonth(), now.getDate(), 12);
        const hour15 = new Date(now.getFullYear(), now.getMonth(), now.getDate(), 15);
        const hour18 = new Date(now.getFullYear(), now.getMonth(), now.getDate(), 18);
        const hour21 = new Date(now.getFullYear(), now.getMonth(), now.getDate(), 21);
        const hour24 = new Date(now.getFullYear(), now.getMonth(), now.getDate(), 24);

        const todaysEvents = events.filter(jumpEvent => new Date(jumpEvent.timeStamp) > startOfDay);

        const time0to3Count = todaysEvents.filter(jumpEvent => new Date(jumpEvent.timeStamp) <= hour3).length;
        const time3to6Count = todaysEvents.filter(jumpEvent => new Date(jumpEvent.timeStamp) > hour3 && new Date(jumpEvent.timeStamp) <= hour6).length;
        const time6to9Count = todaysEvents.filter(jumpEvent => new Date(jumpEvent.timeStamp) > hour6 && new Date(jumpEvent.timeStamp) <= hour9).length;
        const time9to12Count = todaysEvents.filter(jumpEvent => new Date(jumpEvent.timeStamp) > hour9 && new Date(jumpEvent.timeStamp) <= hour12).length;
        const time12to15Count = todaysEvents.filter(jumpEvent => new Date(jumpEvent.timeStamp) > hour12 && new Date(jumpEvent.timeStamp) <= hour15).length;
        const time15to18Count = todaysEvents.filter(jumpEvent => new Date(jumpEvent.timeStamp) > hour15 && new Date(jumpEvent.timeStamp) <= hour18).length;
        const time18to21Count = todaysEvents.filter(jumpEvent => new Date(jumpEvent.timeStamp) > hour18 && new Date(jumpEvent.timeStamp) <= hour21).length;
        const time21to24Count = todaysEvents.filter(jumpEvent => new Date(jumpEvent.timeStamp) > hour21 && new Date(jumpEvent.timeStamp) <= hour24).length;

        let data = [
            createData('00:00', time0to3Count),
            createData('03:00', time3to6Count),
            createData('06:00', time6to9Count),
            createData('09:00', time9to12Count),
            createData('12:00', time12to15Count),
            createData('15:00', time15to18Count),
            createData('18:00', time18to21Count),
            createData('21:00', time21to24Count),
            createData('24:00', 0),
            ];

        jumpData = data;
    }

    return (
        <React.Fragment>
        <Title>Peak Times</Title>
        <ResponsiveContainer>
            <LineChart
            data={jumpData}
            margin={{
                top: 16,
                right: 16,
                bottom: 0,
                left: 24,
            }}
            >
            <XAxis dataKey="time" stroke={theme.palette.text.secondary}>
                Hour
            </XAxis>
            <YAxis stroke={theme.palette.text.secondary}>
                <Label
                angle={270}
                position="left"
                style={{ textAnchor: 'middle', fill: theme.palette.text.primary }}
                >
                Jump Hits
                </Label>
            </YAxis>
            <Line type="monotone" dataKey="amount" stroke={theme.palette.primary.main} dot={false} />
            </LineChart>
        </ResponsiveContainer>
        </React.Fragment>
    );
}

const mapStateToProps = (state) => {
    return {
        demoLocation: state.demoLocation
    };
  }
  
const mapDispatchToProps = (dispatch) => {
    return {
        selectView: (view) => {dispatch({type: 'SELECT_VIEW', view: view})}
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ChartTodayUsage);