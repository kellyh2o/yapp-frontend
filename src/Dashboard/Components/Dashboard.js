import React, { Component } from "react";
import { connect } from "react-redux";
import { JumpsActions } from '../../Jumps/Store';
import { fetchJumps } from '../../Jumps/Store/jumps-actions';
import { fetchUsers } from '../../Users/Store/users-actions';

class Dashboard extends Component { 
    render() {
      this.props.getJumps();
      this.props.getUsers();
      return (
        <div>
          Dashboard Component
        </div>
      )
    }
}

const mapStateToProps = (state) => {
  return {
    posts: state.posts
  };
}

const mapDispatchToProps = (dispatch) => ({
  getJumps: () => {
    dispatch(fetchJumps.request());
  },
  getUsers: () => {
    dispatch(fetchUsers.request());
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);