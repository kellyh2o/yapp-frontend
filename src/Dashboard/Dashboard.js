import React from "react";
import { connect } from "react-redux";
import { LocationsActions } from '../Locations/Store';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';

function Dashboard (props) {

  function generateLocationName() {
    return props.demoLocation ? props.demoLocation.name : "";
  }

  function generateJumpList() {
    if (props.demoLocation && props.demoLocation.jumps) {
      return ( 
        props.demoLocation.jumps.map(element => {
            return (
              <ListItem key={element._id}>
                {element.name}
              </ListItem>
            );
          })
      );
    }
    else {
      return <div />;
    }
  }

  return (
    <div>
      <Box>
        <Typography variant="h4">
          {generateLocationName()}
        </Typography>
      </Box>
      <List>
        {generateJumpList()}
      </List>
    </div>
  )
}

const mapStateToProps = (state) => {
  return {
    token: state.token,
    locations: state.locations,
    demoLocation: state.demoLocation
  };
}

const mapDispatchToProps = (dispatch) => ({
  getLocations: (token) => {
    dispatch(LocationsActions.fetchLocations.request(token));
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);