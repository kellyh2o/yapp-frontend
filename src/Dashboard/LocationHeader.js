import React from 'react';
import { connect } from "react-redux";
import Typography from '@material-ui/core/Typography';
import Title from './Title';

const LocationHeader = (props) => {
  return (
    <React.Fragment>
      <Title>Location</Title>
      <Typography component="p" variant="h4" style={{paddingTop: "20px"}}>
      {props.demoLocation ? props.demoLocation.name : ""}
      </Typography>
    </React.Fragment>
  );
}

const mapStateToProps = (state) => {
  return {
      demoLocation: state.demoLocation
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
      selectView: (view) => {dispatch({type: 'SELECT_VIEW', view: view})}
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(LocationHeader);