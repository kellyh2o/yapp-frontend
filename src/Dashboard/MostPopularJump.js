import React from 'react';
import { connect } from "react-redux";
import { makeStyles } from '@material-ui/core/styles';
import Typography from '@material-ui/core/Typography';
import Title from './Title';

const useStyles = makeStyles({
  depositContext: {
    flex: 2,
  },
});

function getMostPopularJump(demoLocation) {
  if (demoLocation && demoLocation.jumps) {
    const jumps = demoLocation.jumps;
    const orderedJumps = jumps.sort((x, y) => {
      return y.events.length - x.events.length;
    });
  
    return orderedJumps[0].name;
  } else {
    return "- - -";
  }
}

function getPopularJumpNumberOfHits(demoLocation) {
  if (demoLocation && demoLocation.jumps) {
    const jumps = demoLocation.jumps;
    const orderedJumps = jumps.sort((x, y) => {
      return y.events.length - x.events.length;
    });
  
    return orderedJumps[0].events.length;
  } else {
    return "- - -";
  }
}


const MostPopularJump = (props) => {
  const classes = useStyles();
  return (
    <React.Fragment>
      <Title>Today's Most Popular Jump</Title>
      <Typography component="p" variant="h4" style={{paddingTop: "20px"}}>
        {getMostPopularJump(props.demoLocation)}
      </Typography>
      <Typography color="textSecondary" className={classes.depositContext} style={{paddingTop: "20px", paddingLeft: "5px"}}>
        {getPopularJumpNumberOfHits(props.demoLocation)} Hits Today
      </Typography>
    </React.Fragment>
  );
}

const mapStateToProps = (state) => {
  return {
      demoLocation: state.demoLocation
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
      selectView: (view) => {dispatch({type: 'SELECT_VIEW', view: view})}
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(MostPopularJump);