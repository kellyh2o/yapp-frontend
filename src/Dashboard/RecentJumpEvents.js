import React from 'react';
import { connect } from "react-redux";
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import Title from './Title';
import moment from 'moment';

function buildTableData(demoLocation) {

    let tableData = [];
        
    if (demoLocation && demoLocation.jumps) {
        const jumps = demoLocation.jumps;

        jumps.forEach(jump => {
            jump.events.forEach(jumpEvent => {

                let customId = `${jump._id}-----${jumpEvent._id}`;

                if (tableData.filter(data => data.id === customId).length === 0) {
                    tableData.push({
                        id: customId,
                        date: new Date(jumpEvent.timeStamp),
                        jump: jump.name,
                        riderMass: jumpEvent.riderMass,
                        riderSpeed: jumpEvent.riderSpeed
                    });
                }
            });
        });
    }

    return tableData.sort((x, y) => {
        return y.date - x.date;
    });
}

const RecentJumpEvents = (props) => {

    let tableData = buildTableData(props.demoLocation);

    return (
        <React.Fragment>
        <Title>Recent Jump Events</Title>
        <Table size="small">
            <TableHead>
            <TableRow>
                <TableCell>Date</TableCell>
                <TableCell>Jump Name</TableCell>
                <TableCell>Rider Mass (lbs) </TableCell>
                <TableCell>Rider Speed (mph) </TableCell>
            </TableRow>
            </TableHead>
            { tableData && tableData.length ? (
                <TableBody>
                {tableData.map(data => (
                    <TableRow key={data.id}>
                        <TableCell>{moment(data.date.toString()).calendar()}</TableCell>
                        <TableCell>{data.jump}</TableCell>
                        <TableCell>{data.riderMass}</TableCell>
                        <TableCell>{data.riderSpeed}</TableCell>
                    </TableRow>
                ))}
                </TableBody>)
             : <TableBody />
            }
        </Table>
        </React.Fragment>
    );
}

const mapStateToProps = (state) => {
    return {
        demoLocation: state.demoLocation
    };
  }
  
  const mapDispatchToProps = (dispatch) => {
    return {
        selectView: (view) => {dispatch({type: 'SELECT_VIEW', view: view})}
    }
  }
  
  export default connect(mapStateToProps, mapDispatchToProps)(RecentJumpEvents);