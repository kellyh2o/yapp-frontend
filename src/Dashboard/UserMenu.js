import React from 'react';
import Button from '@material-ui/core/Button';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import Grow from '@material-ui/core/Grow';
import Paper from '@material-ui/core/Paper';
import Popper from '@material-ui/core/Popper';
import MenuItem from '@material-ui/core/MenuItem';
import MenuList from '@material-ui/core/MenuList';
import { makeStyles } from '@material-ui/core/styles';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';
import IconButton from '@material-ui/core/IconButton';
import { connect } from 'react-redux';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
  },
  paper: {
    marginRight: theme.spacing(2),
  },
}));

const UserMenu = (props) => {
  const classes = useStyles();
  const [open, setOpen] = React.useState(false);
  const anchorRef = React.useRef(null);

  const handleToggle = () => {
    setOpen((prevOpen) => !prevOpen);
  };

  const handleClose = (event) => {
    if (anchorRef.current && anchorRef.current.contains(event.target)) {
      return;
    }

    setOpen(false);
  };

  function handleListKeyDown(event) {
    if (event.key === 'Tab') {
      event.preventDefault();
      setOpen(false);
    }
  }

  // return focus to the button when we transitioned from !open -> open
  const prevOpen = React.useRef(open);
  React.useEffect(() => {
    if (prevOpen.current === true && open === false) {
      anchorRef.current.focus();
    }

    prevOpen.current = open;
  }, [open]);

  return (
    <div className={classes.root}>
      <div>
        <IconButton
          color="inherit"
          ref={anchorRef}
          aria-controls={open ? 'menu-list-grow' : undefined}
          aria-haspopup="true"
          onClick={handleToggle}
        >
          <AccountCircleIcon fontSize="large" />
        </IconButton>
        <Popper open={open} anchorEl={anchorRef.current} role={undefined} transition disablePortal placement="bottom-end" >
            <Paper>
                <ClickAwayListener onClickAway={handleClose}>
                <MenuList autoFocusItem={open} id="menu-list-grow" onKeyDown={handleListKeyDown}>
                    <MenuItem onClick={() => { props.selectView('UserProfile')}}>My Account</MenuItem>
                    <MenuItem onClick={() => { props.selectView('Settings')}}>Settings</MenuItem>
                    <MenuItem onClick={handleClose}>Logout</MenuItem>
                </MenuList>
                </ClickAwayListener>
            </Paper>
        </Popper>
      </div>
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
      isDashboardSelected: state.selectedView === "Dashboard",
      isUserProfileSelected: state.selectedView === "UserProfile",
      isSettingsSelected: state.selectedView === "Settings",
  };
}

const mapDispatchToProps = (dispatch) => {
  return {
      selectView: (view) => {dispatch({type: 'SELECT_VIEW', view: view})}
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(UserMenu);