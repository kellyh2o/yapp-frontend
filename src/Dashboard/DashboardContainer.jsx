import React from "react";
import { connect } from "react-redux";
import { LocationsActions } from '../Locations/Store';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import Dashboard from './Dashboard';

const DashboardContainer = (props) => {

  function generateLocationName() {
    return props.demoLocation ? props.demoLocation.name : "";
  }

  function generateJumpList() {
    if (props.demoLocation && props.demoLocation.jumps) {
      return ( 
        props.demoLocation.jumps.map(element => {
            return (
              <ListItem key={element._id}>
                {element.name}
              </ListItem>
            );
          })
      );
    }
    else {
      return <div />;
    }
  }

  return (
    <Dashboard />
    // <div style={{paddingLeft: "20px", width: "400px"}}>
    //   <Box>
    //     <Typography variant="h4">
    //       {generateLocationName()}
    //     </Typography>
    //   </Box>
    //   <List>
    //     {generateJumpList()}
    //   </List>
    // </div>
  )
}

const mapStateToProps = (state) => {
  return {
    token: state.token,
    locations: state.locations,
    demoLocation: state.demoLocation
  };
}

const mapDispatchToProps = (dispatch) => ({
  getLocations: (token) => {
    dispatch(LocationsActions.fetchLocations.request(token));
  }
})

export default connect(mapStateToProps, mapDispatchToProps)(DashboardContainer);