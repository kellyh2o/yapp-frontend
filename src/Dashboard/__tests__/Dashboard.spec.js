import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import Dashboard from "../Dashboard";

describe('Dashboard', () => {
    it("should render", () => {
        const mockStore = configureMockStore();

        const initialState = {
            demoLocation: {
                name: "Hogwarts",
                jumps: [ 
                    {
                        name: "Jump1",
                        events: []
                    },
                    {
                        name: "Jump2",
                        events: []
                    },
                    {
                        name: "Jump3",
                        events: [
                            {
                                riderMass: 22,
                                riderSpeed: 12,
                            }
                        ]
                    }
                ]
            }
        };
        const store = mockStore(initialState);

        const { asFragment } = render(
            <Provider store={store}>
                <Dashboard />
            </Provider>
        );

        expect(asFragment()).toMatchSnapshot();
    })
});

