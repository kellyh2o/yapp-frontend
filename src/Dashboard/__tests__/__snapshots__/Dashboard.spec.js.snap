// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      style="margin-bottom: 20px; padding: 20px;"
    >
      <h2
        class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
      >
        Location
      </h2>
      <p
        class="MuiTypography-root MuiTypography-h4"
        style="padding-top: 20px;"
      >
        Hogwarts
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-9"
      >
        <div
          class="MuiPaper-root makeStyles-paper-1 makeStyles-fixedHeight-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
          >
            Peak Times
          </h2>
          <div
            class="recharts-responsive-container"
            style="width: 100%; height: 100%;"
          >
            <div
              style="position: absolute; width: 0px; height: 0px; visibility: hidden; display: none;"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3"
      >
        <div
          class="MuiPaper-root makeStyles-paper-1 makeStyles-fixedHeight-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
          >
            Today's Most Popular Jump
          </h2>
          <p
            class="MuiTypography-root MuiTypography-h4"
            style="padding-top: 20px;"
          >
            Jump3
          </p>
          <p
            class="MuiTypography-root makeStyles-depositContext-164 MuiTypography-body1 MuiTypography-colorTextSecondary"
            style="padding-top: 20px; padding-left: 5px;"
          >
            1 Hits Today
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <h2
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
          >
            Recent Jump Events
          </h2>
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Date
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Jump Name
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Rider Mass (lbs) 
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall"
                  scope="col"
                >
                  Rider Speed (mph) 
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                >
                  Invalid date
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                >
                  Jump3
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                >
                  22
                </td>
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                >
                  12
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
